### DISCUSSION TITLE
Reading a Yaml file as a part of a MOOSE input

### DISCUSSION THREAD

**Discussion by gsgall** [idaholab/moose#25809] [Q&A General] [answered]
I am interested in using a Yaml file to hold some information about a chemical reaction network. I would like to parse this file and then create kernels and material properties based on the results of this parsing. Is there already a standard or recommended method for reading Yaml files as a part of input files?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Unfortunately we do not have a YAML reader user object. We only have CSV and JSON right now. 
This would have been the best way to do this, with the user object reading the file and any other object being able to query it for information. 

We have a YAML formatter for output that's it. 
You ll have to start by writing (or finding an external package) code to read the YAML file. Then you should make a YAMLReader user object that reads and has routines for other objects to query infomration from. Then you can simply retrieve the user object from any object using its name. There is a UserObjectInterface that most objects inherit and that defines:
```
const T & getUserObject(const std::string & param_name, bool is_dependency = true) const;
```

Guillaume

**Reply by gsgall** [reply]
Thank you so much!
